social_distribution/posts/views.py

```diff
@@ -57,7 +57,17 @@ def post_create(request, author_id):
                 friends = author.followers.all() & author.followings.all()
                 for friend in friends:
                     friend.inbox.posts.add(post)
-            # TODO: inbox private posts
+            elif post.visibility == "private":
+                friends = author.followers.all() & author.followings.all()
+                # if users are friend
```

Comment:
- Friends are also followers so  if you do 2 for loops like this, there will inherently be duplicates.
- It is not very optimal to traverse through the whole queryset, and check if the usernames are matching, just get author object of the recipient and check if it is in the followers list.